samza-kafka/src/main/java/org/apache/samza/system/kafka/KafkaSystemConsumer.java

```diff
@@ -167,12 +142,11 @@ public void start() {
 
   private void startSubscription() {
     //subscribe to all the registered TopicPartitions
-    Set<TopicPartition> registeredTopicPartitions = new HashSet<>(topicPartitionToStartpointMap.keySet());
-    LOG.info("{}: Consumer subscribes to {}", this, registeredTopicPartitions);
+    LOG.info("{}: Consumer subscribes to {}", this, topicPartitionsToSSP.keySet());
     try {
       synchronized (kafkaConsumer) {
         // we are using assign (and not subscribe), so we need to specify both topic and partition
-        kafkaConsumer.assign(registeredTopicPartitions);
+        kafkaConsumer.assign(topicPartitionsToSSP.keySet());
```

Comment:
Do you think it'll be beneficial to use the `ThreadSafeKafkaConsumer` wrapper class? Maybe as part of the follow up PR?